0xProject/0x-api
# chore: add slippage protection flag to quote reports [GRO-28]

# Description


Data uses Quote Reports to compare settled trades versus the quotes.

We need to add a field to the quote reports to indicate if a quote was slippage protected. Quotes are slippage protected by default unless the enableSlippageProtection parameter is set to false. Being able to distinguish slippage-protected trades is important for analyzing the impact of the feature.

This change affects quote generated in swap_handlers.ts and meta_transaction_service.ts. Note that the meta transactions are not slippage protected, so the flag will always be false.

# Testing Instructions



Setup kafka locally and configure 0xAPI to point to mainnet and your local kafka. Run some quote requests against your local endpoint and verify the quote reports sent to kafka.

More information on kafka can be found at:
https://www.conduktor.io/kafka/starting-kafka

# Checklist





    For more information see `0x API Matcha smoke test runbook` in Quip.
